packages/central-server/src/apiV2/utilities/translateQuestionDependentFields.js

```diff
@@ -5,40 +5,51 @@ import { splitStringOn, splitStringOnComma } from './split';
  * to the keys that will be used in the output
  */
 const ENTITY_FIELD_TRANSLATION = {
-  name: 'name',
-  code: 'code',
   parent: 'parentId',
   grandparent: 'grandparentId',
 };
 
 // Subfields will be in the format field.subfield, e.g. 'attributes.type'
-const isSubfield = (nestedFieldList, fieldKey) =>
+const isNested = (nestedFieldList, fieldKey) =>
   nestedFieldList.find(field => fieldKey.startsWith(field));
 
+const translateNestedField = (rawFieldKey, rawFieldValue) => {
+  const splitField = splitStringOn(rawFieldKey, '.');
```

Comment:
@rohan-bes just wanted to get your take on this redo. It's generic now, and I've split out functions a little.